CONTRIBUTING.md

```diff
@@ -1,50 +1,125 @@
+- [Release 🛳](#release-%F0%9F%9B%B3)
   - [Main version](#main-version)
   - [Maintenance version](#maintenance-version)
-- [Update docs](#update-docs)
+- [Updating the docs 📚](#updating-the-docs-)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
-[![InstantSearch.js logo][logo]][website]
+## Contributing with an new issue ⚠️
 
-A note about the branches used in this project:
- - `develop` is for the major version in active development (currently v2)
- - `maintenance` is for the previous major version (currently v1)
- - `master` is for the main current stable version
+Opening an issue is very useful for us so don't hesitate, we'll make sure to fix it
+as soon as possible if it's technically feasible and doesn't have important side effects
+for the other users.
 
-You should do the dev and PR according to the target version. No new features
-will be done on the maintenance version.
+Before opening an issue, first check that there is not an already open issue for the same
+topic using the [issues tab](https://github.com/algolia/instantsearch.js/issues). Don't
+hesitate to thumb up an issue that corresponds to the problem you found.
 
-Once you make a pull request, a bot will comment with a link to a development version of the website. On it you can find
+Another element that will help us go faster at solving the issue is to provide a reproducible
+test case. We often recommend to [use this jsFiddle template](https://jsfiddle.net/bobylito/9h7sgo10/).
 
-1. The generated documentation
+## The code contribution process 🤘
+
+On your side:
+ - Fork and clone the project
+ - Create a new branch for what you want to solve (fix/*issue number*, feat/2.x-*name of the feature*)
+ - Make your changes
+ - Open a pull request
+
+Depending on what you're working, you might consider different base branches.
+
+Then:
+ - Peer review of the pull request (by at least one of the core contributors)
+ - Automatic checks ([tests](#tests-), [commits](#commit-conventions-), [linters](#lint-))
+ - When everything is green, your contribution is merged 🚀
+
+After you create a pull request, a bot will comment with a link to a development version of the website.
+
+On it you can find:
+1. The generated documentation: https://**deploy-url-netlify**/
     - for example: https://deploy-preview-2383--algolia-instantsearch.netlify.com/
     - source: https://github.com/algolia/instantsearch.js/tree/develop/docgen
-2. A playground for the widgets
+2. A playground for the widgets: https://**deploy-url-netlify**/dev-novel
     - for example: https://deploy-preview-2383--algolia-instantsearch.netlify.com/dev-novel/
     - source: https://github.com/algolia/instantsearch.js/tree/develop/dev
 
-### Requirements
+## Commit conventions 🤓
+
+This project follows the [conventional changelog](https://conventionalcommits.org/) approach.
+This means that all commit messages should be formatted using the following scheme:
+
+```
+<type>[<scope>]: <description>
```

Comment:
I would write this as: 

```
type(scope): description
```

Where scope can be left out


Because not everyone knows the convention of having brackets for optional and parens etc.

Reply:
I reused the syntax of the [RFCish document](https://conventionalcommits.org/) about the conventional commits. That being your approach is more friendly and thus, will make the document more approachable.